### ISSUE TITLE
[Emterpreter] Cant get EMTERPRETIFY_FILE to work with NodeJS

### ISSUE THREAD

**Issue by Mashewnutz** [kripken/emscripten#4240]
Hey,

I'm trying to run from NodeJS using Emterpreter and EMTERPRETIFY_FILE flag set. From the notes in settings.js I found this:

"If you emit JS, you need to make sure that Module.emterpreterFile contains an ArrayBuffer with the byte code"

Because I'm running from Node, I think I have to load the file myself. So I'm using --pre-js preload.js option to load and set the property on Module because the rest of the code runs.

The contents of my preload.js is:

var Module = {};

var fs = require('fs');
fs.readFile( __dirname + '/code.binary', function (err, data) {
  if (err) {
    throw err; 
  }  
  Module.emterpreterFile = data.buffer;
});

But the readFile function never seems to execute... Am I missing something or is there an easier way to make this work with NodeJS

Thanks,
Mash

**Comment by Mashewnutz** [comment]
OK I managed to solve this my using fs.readFileSync.

However, after all that, I haven't notice any differences in startup time, OR build times. In fact it takes LONGER to build.

Am I doing something wrong?

**Comment by kripken** [comment]
First, it depends on the program. The startup improvement would be most noticeable on large projects, where a lot of code is replaced by a binary file.

Second, it depends on the JS engine. The latest version of v8 should use turbofan which does some aggressive compilation of asm.js code, and the emterpreter option can avoid that. But on older versions of v8, the difference might be much smaller. So this might depend on what version of v8 your node uses.

**Comment by Mashewnutz** [comment]
OK, the warning I got was this:

"warning: emterpreter bytecode is fairly large, 5.41 MB [...]"

5.41MB doesn't sound particularly large to me so I'm guessing thats why I didn't see any speed ups.

I tried running this to see which version of V8 was being used:
matthew.casey@cpp: node -e 'console.log(process.versions.v8);'
4.5.103.33

Anyway, I managed to get a huge speed up yesterday by adding -O2! My tests previously running in at 44s went down to 4s! 

Amazing!
